LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/the-img-element/decode/image-decode-path-changes-svg.tentative-expected.txt

```diff
@@ -1,8 +1,8 @@
 
-FAIL SVGImageElement.prototype.decode(), href mutation tests. xlink:href changes fail decode. img.decode is not a function. (In 'img.decode()', 'img.decode' is undefined)
-FAIL SVGImageElement.prototype.decode(), href mutation tests. href changes fail decode. img.decode is not a function. (In 'img.decode()', 'img.decode' is undefined)
-FAIL SVGImageElement.prototype.decode(), href mutation tests. xlink:href changes fail decode; following good decode succeeds. img.decode is not a function. (In 'img.decode()', 'img.decode' is undefined)
-FAIL SVGImageElement.prototype.decode(), href mutation tests. href changes fail decode; following good decode succeeds. img.decode is not a function. (In 'img.decode()', 'img.decode' is undefined)
-FAIL SVGImageElement.prototype.decode(), href mutation tests. xlink:href changes fail decode; following bad decode fails. img.decode is not a function. (In 'img.decode()', 'img.decode' is undefined)
-FAIL SVGImageElement.prototype.decode(), href mutation tests. href changes fail decode; following bad decode fails. img.decode is not a function. (In 'img.decode()', 'img.decode' is undefined)
+FAIL SVGImageElement.prototype.decode(), href mutation tests. xlink:href changes fail decode. assert_unreached: Should have rejected: undefined Reached unreachable code
+FAIL SVGImageElement.prototype.decode(), href mutation tests. href changes fail decode. assert_unreached: Should have rejected: undefined Reached unreachable code
+FAIL SVGImageElement.prototype.decode(), href mutation tests. xlink:href changes fail decode; following good decode succeeds. assert_unreached: Should have rejected: undefined Reached unreachable code
+FAIL SVGImageElement.prototype.decode(), href mutation tests. href changes fail decode; following good decode succeeds. assert_unreached: Should have rejected: undefined Reached unreachable code
```

Comment:
Honestly, I haven't investigated much because I was thinking of it as stub implementation.

Reply:
If it's a stub implementation it needs to be behind a preference. Definitely if you're not sure about it.

Reply:
I just closed PR for future for someone to take care of it. :-)

Reply:
These tests are failing because changing the image source should cause the decoding promise to be rejected. But currently it is resolved.

I think this change is fine. With it, `SVGImageElement.decode()` will be working as the `HTMLImageElement.decode()`. I am almost sure that I can reproduce the same errors with `HTMLImageElement`.

So my suggestions are:

1. Reopen this PR and please add at least one layout test for SVGImageElement. You can add a test similar to fast/images/decode-static-image-resolve.html.
2. File a new bug for the remaining issues, like: "Changing the image source does not reject the decode() promise". This bug should cover the `HTMLImageElement` and `SVGImageElement`.

Reply:
@shallawa - Thanks for your input. I have added test as suggested and opened new bug here - https://bugs.webkit.org/show_bug.cgi?id=279623